custom/ilsgateway/tanzania/handlers/la.py

```diff
@@ -41,14 +43,15 @@ def handle(self):
         )
         error = False
         products_without_soh = set()
-        for product_code, quantity in parsed_report:
-            try:
-                product_id = SQLProduct.objects.get(domain=self.domain, code__iexact=product_code).product_id
-                self._create_stock_transaction(report, product_id, quantity)
-            except SQLProduct.DoesNotExist:
-                error = True
-            except StockState.DoesNotExist:
-                products_without_soh.add(product_code.lower())
+        with transaction.atomic():
```

Comment:
i just realized, shouldn't the call to `StockReport.objects.create` above also be within this transaction.atomic() block?